internal/util/util.go

```diff
@@ -292,3 +293,28 @@ func HasBackupLabel(o *metav1.ObjectMeta, backupName string) bool {
 	}
 	return o.Labels[velerov1api.BackupNameLabel] == label.GetValidName(backupName)
 }
+
+func DeleteVolumeSnapshot(vs *snapshotv1api.VolumeSnapshot, log logrus.FieldLogger) error {
+	log.Infof("Deleting Volumesnapshot %s/%s", vs.Namespace, vs.Name)
+	_, snapClient, err := GetClients()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if vs.Status != nil && vs.Status.BoundVolumeSnapshotContentName != nil {
+		// we patch the DeletionPolicy of the volumesnapshotcontent to set it to Delete.
+		// This ensures that the volume snapshot in the storage provider is also deleted.
+		err := SetVolumeSnapshotContentDeletionPolicy(*vs.Status.BoundVolumeSnapshotContentName, snapClient.SnapshotV1())
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, fmt.Sprintf("failed to patch DeletionPolicy of volume snapshot %s/%s", vs.Namespace, vs.Name))
+		}
+
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+	}
+	err = snapClient.SnapshotV1().VolumeSnapshots(vs.Namespace).Delete(context.TODO(), vs.Name, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
this is just cleanup step, I think we can ignore throwing errors here. 
Just log it and ignore it